MetaMask/metamask-mobile
# fix: Rounding issue in approval amount (use site suggestion)

## **Description**



There is an [issue](https://github.com/MetaMask/metamask-mobile/issues/8199) that shows that really large dapp proposed value is rounded off for approve transaction. Noticed that this happens when hexToBN value is wrapped using the Number function. This PR specifies add a boolean to prevent this  for approve transactions while other use cases should act as it is.

## **Related issues**

Fixes: #8199 

## **Manual testing steps**

1. Enable input customisation for erc20 token. [See](https://github.com/MetaMask/metamask-mobile/issues/8199#issuecomment-1886096624)
2. Enter a really large value
3. See it doesn't round off the dapp proposed value

## **Screenshots/Recordings**



### **Before**
https://recordit.co/l0DWAilLla

### **After**
https://recordit.co/gEKmKf9Yq7

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
